eclipse-theia/theia
# Enable context isolation and disable nodejs support. Fixes #2018

#### What it does
All access to electron API is now done through an API exposed via a preload script. Access to the electron API (including electron-remote) and nodejs API is no longer possible.
Theia extensions can contribute to the preload script via a `theiaExtensions` module declaration in their package.json

Fixes #2018

Contributed on behalf or STMicroelectronics

#### How to test
Run the electron version of Theia and make sure nothing is broken. Particular areas of interest is anything to do with menus and windows, for example "zoom in/out", "maximize", etc.

#### CQ

- [x] [Electron IP Check](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7508)
- [x] [FFmpeg IP Check](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7507)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
